bindings/java/src/org/sleuthkit/datamodel/HostAddressManager.java

```diff
@@ -43,7 +46,23 @@
 	private static final Logger LOGGER = Logger.getLogger(HostAddressManager.class.getName());
 
 	private final SleuthkitCase db;
+	
+	/**
+	 * An HostAddress Object Id entry is maintained in this cache when a
+	 * hostaddress and ip mapping is added. This is here to improve the
+	 * performance of {@link #existsHostNameAndIpMapping(long) } check.
+	 */
+	private final Cache<Long, Byte> recentHostNameAndIpMappingCache = CacheBuilder.newBuilder().maximumSize(200000).build();
 
+	/**
+	 * Recently added or accessed Host Address Object Ids are cached. This is
+	 * here to improve performance of the
+	 * {@link #existsHostAddress(org.sleuthkit.datamodel.HostAddress.HostAddressType, java.lang.String)}
+	 * check.
+	 */
+	private final Cache<String, Long> recenHostAddresstCache = CacheBuilder.newBuilder().maximumSize(200000).build();
```

Comment:
The 't' seems misplaced (should be recentHostAddressCache, right?)

Reply:
tough audience. Fixed